### PR TITLE
fix: do not set document.referrer while using cy.visit

### PR DESCRIPTION
**Pull request by klarzynskik**
- Closes https://github.com/cypress-io/cypress/issues/4295

### User facing changelog
- `document.referrer` would no longer be set while using cy.visit

### Additional details
While using the `cy.visit`, the `document.referrer` was set incorrectly to the Cypress test URL, while it should be an empty string in such cases.

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [n/a] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by davidmunechika — APPROVED** [review]

File: packages/driver/src/cy/commands/navigation.ts [hunk 1]
```diff
@@ -789,6 +789,15 @@ export default (Commands, Cypress, cy, state, config) => {
 
           const onBeforeLoad = (contentWindow) => {
             try {
+              // when using the visit the document referrer should be set to an empty string
+              if (config('modifyObstructiveCode')) {
```
Comment (flotwig) [inline packages/driver/src/cy/commands/navigation.ts:793]: This doesn't need to be behind `modifyObstructiveCode`. `modifyObstructiveCode` controls if the `regex-rewriter` in the proxy layer runs or not, which is unrelated to this change to `document.referrer`. I think this fix is fine to apply to all `cy.visit()`s.

**Review by flotwig — CHANGES_REQUESTED** [review]

File: packages/driver/src/cy/commands/navigation.ts (hunk 1 above)
Comment (klarzynskik) [reply packages/driver/src/cy/commands/navigation.ts:793]: this was suggested by @jennifer-shehane in the original issue description, will change 👍🏼

Comment (klarzynskik) [reply packages/driver/src/cy/commands/navigation.ts]: @flotwig fixed in https://github.com/cypress-io/cypress/pull/18658/commits/568e9e58366aa934270b26c928c4794f7b2df700

**Review by flotwig — APPROVED** [review]
LGTM. Thank you for the contribution @klarzynskik

**Pull request merged**